### DISCUSSION TITLE
How to translate things outside a component

### DISCUSSION THREAD

**Discussion by DennisVanIngen** [robisim74/qwik-speak#75] [Q&A]
I want to translate my error messages that come from the server with modular forms like this: 
```js
throw new FormError<LoginForm>({
  email: 'This email has been blacklisted.',
});
```
But also in zod when I do this:
```js
z.object({
  manufacturerId: z.string().min(1, 'Select a manufacturer'),
});
```

**Comment by robisim74** [comment]
Please, see https://github.com/robisim74/qwik-speak/issues/61

To translate you need the context, so in case it is not even possible to pass the context and use [inlineTranslate](https://robisim74.gitbook.io/qwik-speak/library/translate#inlinetranslate), it is necessary to translate the messages where they are displayed, that is in `component$`

Of course you need to map the messages coming from the server into the translation files.

**Comment by robisim74** [comment]
From v0.17.0, the new `inlineTranslate` with no context is better suited to translation out of components.

See https://robisim74.gitbook.io/qwik-speak/library/translate and https://robisim74.gitbook.io/qwik-speak/library/translate#server-translation
